plugins/UFM_NDT_Plugin/ufm_sim_web_service/topo_diff/ndt_infra.py

```diff
@@ -0,0 +1,316 @@
+IBDIAGNET_AGE_INTERVAL_DEFAULT = 3600
+MERGER_OPEN_SM_CONFIG_FILE = "%s/%s" % (TOPOCONFIG_DIRECTORY, "topoconfig")
+LAST_DEPLOYED_NDT_FILE_INFO = "%s/%s" % (TOPOCONFIG_DIRECTORY, "deployed_ndt")
+SWITCH_GUID = "Switch_GUID" 
+PORT_NUM = "Port_Num" 
+NEIGHBOR_GUID = "Neighbor_GUID" 
+NEIGHBOR_PORT_NUM = "Neighbor_Port_Num" 
+NEIGHBOR_TYPE = "Neighbor_Type"
+PORT_STATE = "Port_State"
+BOUNDARY_PORT_STATE_NO_DISCOVER = "No-discover"
+BOUNDARY_PORT_STATE_DISABLED = "Disabled"
+BOUNDARY_PORT_STATE_ACTIVE = "Active"
+BOUNDARY_PORTS_STATES = [BOUNDARY_PORT_STATE_NO_DISCOVER,
+                         BOUNDARY_PORT_STATE_DISABLED,
+                         BOUNDARY_PORT_STATE_ACTIVE]
+TOPOCONFIG_FIELD_NAMES = ["port_guid", "port_num", "peer_port_guid", "peer_port_num", "host_type", "port_state"]
+NDT_FILE_STATE_NEW = "new"
+NDT_FILE_STATE_VERIFIED = "verified"
+NDT_FILE_STATE_DEPLOYED = "deployed"
+NDT_FILE_STATE_UPDATED = "updated"
+NDT_FILE_STATE_UPDATED_DISABLED = "updated_disabled"
+NDT_FILE_STATE_UPDATED_NO_DISCOVER = "updated_no_discover"
+NDT_FILE_STATE_DEPLOYED_DISABLED = "deployed_disabled"
+NDT_FILE_STATE_DEPLOYED_NO_DISCOVER = "deployed_no_discover"
+
+def run_command_line_cmd(command):
+    cmd = command.split()
+    p = subprocess.Popen(cmd,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.STDOUT)
+    return iter(p.stdout.readline, b'')
+
+def execute_generic_command_interractive(command):
+    logging.info("Server:Executing Command %s" % command)
+    try:
+        return run_command_line_cmd(command)
+    except Exception as e:
+        logging.error("Got exception when executing command")
+        logging.error(e)
+        return "Error on command %s execution %s command: %s" % (command,e)
+
+def execute_generic_command(command):
+    logging.info("Executing Command %s" % command)
+    try:
+        process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE,
+                                                        stderr=subprocess.PIPE)
+        output, error = process.communicate()
+        logging.debug("Command Errors: %s" % error)
+        retcode = process.returncode
+        if retcode != 0:
+            logging.error(
+                "Executing command failed: %s, status %s" % (
+                    command, retcode))
+            return False, "Failed to execute command %s command: %s" % (command,
+                                                                         retcode)
+    except Exception as e:
+        logging.error("Got exception when executing command")
+        logging.error(e)
+        return False,"Error on command %s execution : %s" % (command,e)
+    ret_msg = EMPTY_RESPOND_MESSAGE if not output else output
+    return True, ret_msg
+
+def check_file_exist(file_name):
+    '''
+    check if file exist
+    :param file_name:
+    '''
+    if not os.path.isfile(file_name):
```

Comment:
return os.path.isfile(file_name)
do you really need this function?

Reply:
I need it :)